src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -717,6 +717,42 @@ export const updateFormSettings = (
   })
 }
 
+/**
+ * Creates form logic.
+ * @param form The original form to create logic in
+ * @param createdLogic Object containing the created logic
+ * @returns ok(created logic dto) on success
+ * @returns err(database errors) if db error is thrown during logic update
+ */
+export const createFormLogic = (
+  form: IPopulatedForm,
+  createdLogic: LogicDto,
+): ResultAsync<ILogicSchema, DatabaseError | FormNotFoundError> => {
+  // Create new form logic
+  return ResultAsync.fromPromise(
+    FormModel.createFormLogic(form._id.toHexString(), createdLogic),
```

Comment:
why do we invoke `toHexString()` here but not when logging?

Reply:
thanks! I guess we could just not cast to string for both to be consistent, edited accordingly